modules/gapi/include/opencv2/gapi/render.hpp

```diff
@@ -78,35 +83,96 @@ struct Line
     int        thick;  //!< The thickness of line
     int        lt;     //!< The Type of the line. See #LineTypes
     int        shift;  //!< The number of fractional bits in the point coordinates
+};
+
+/**
+ * A structure to represent parameters for drawing a mosaic
+ */
+struct Mosaic
+{
+    cv::Rect   mos;    //!< Coordinates of the mosaic
+    int        cellSz; //!< Cell size (same for X, Y). Note: mos size must be multiple of cell size
+    int        decim;  //!< Decimation (0 stands for no decimation)
+};
+
+/**
+ * A structure to represent parameters for drawing an image
+ */
+struct Image
+{
+    int x;          //!< x-coord of image
+    int y;          //!< y-coord of image
+    cv::Mat img;    //!< Image to draw
+    cv::Mat alpha;  //!< Alpha channel for image to draw (same size and number of channels)
```

Comment:
Alpha channel is usually a grayscale image (mask), why it has to be of the same number of channels here?

Reply:
@TolyaTalamanov any point on this>

Reply:
@TolyaTalamanov, could you, please, fix it? Let's discuss locally what was the problem here.